moler/device/proxy_pc2.py

```diff
@@ -0,0 +1,396 @@
+# f.ex. moler.device.textualdevice introduces state TextualDevice.not_connected = "NOT_CONNECTED"
+NOT_CONNECTED = TextualDevice.not_connected
+CONNECTION_HOPS = TextualDevice.connection_hops
+UNIX_LOCAL = UnixLocal.unix_local
+UNIX_LOCAL_ROOT = UnixLocal.unix_local_root
+PROXY_PC = "PROXY_PC"
+
+
+def want_local_unix_state(io_type=None, io_connection=None):
+    """
+    Check if device is intended to work with local machine or remote ones only.
+    :return: True for local.
+    """
+    if io_type == "terminal":
+        return True
+    if (platform.system() != 'Windows') and isinstance(io_connection, ThreadedTerminal):
+        return True
+    else:  # all remote-access connections (tcp, udp, telnet, ssh); even connecting to localhost
+        return False
+
+
+@six.add_metaclass(abc.ABCMeta)
+class ProxyPc2(UnixLocal):
+
+    def __init__(self, sm_params, name=None, io_connection=None, io_type=None, variant=None, io_constructor_kwargs=None,
+                 initial_state=None, lazy_cmds_events=False):
+        """
+        Create Unix device communicating over io_connection
+        :param sm_params: dict with parameters of state machine for device
+        :param name: name of device
+        :param io_connection: External-IO connection having embedded moler-connection
+        :param io_type: type of connection - tcp, udp, ssh, telnet, ...
+        :param variant: connection implementation variant, ex. 'threaded', 'twisted', 'asyncio', ...
+                        (if not given then default one is taken)
+        :param io_constructor_kwargs: additional parameter into constructor of selected connection type
+                        (if not given then default one is taken)
+        :param initial_state: name of initial state. State machine tries to enter this state just after creation.
+        :param lazy_cmds_events: set False to load all commands and events when device is initialized, set True to load
+                        commands and events when they are required for the first time.
+        """
+        self._use_local_unix_state = want_local_unix_state(io_type, io_connection)
+        base_state = UNIX_LOCAL if self._use_local_unix_state else NOT_CONNECTED
+        self._use_proxy_pc = self._should_use_proxy_pc(sm_params, PROXY_PC)
+        base_or_final_state = PROXY_PC if self._use_proxy_pc else base_state
+        initial_state = initial_state if initial_state is not None else base_or_final_state
+        super(ProxyPc2, self).__init__(name=name, io_connection=io_connection,
+                                       io_type=io_type, variant=variant,
+                                       io_constructor_kwargs=io_constructor_kwargs,
+                                       sm_params=sm_params, initial_state=initial_state,
+                                       lazy_cmds_events=lazy_cmds_events)
+        self._prompt_detector_timeout = 0.5
+        self._after_open_prompt_detector = None
```

Comment:
What do you think to add to all `__init__` for devices log with information like "It's an experimental implementation of ssh device. This functionality will be merged to other device in Moler 2.0.0 and discontinued from this class" ? (or something like this)

Reply:
good idea

Reply:
Solution shows following warnings in logs:

```
Class ProxyPc2 is an experimental/temporary implementation of device utilizing sshshell connection. It's functionality will be merged into ProxyPc device in Moler 2.0.0 and ProxyPc2 will be removed.

Class UnixRemote2 is an experimental/temporary implementation of device utilizing sshshell connection. It's functionality will be merged into UnixRemote device in Moler 2.0.0 and UnixRemote2 will be removed.

Class AdbRemote2 is an experimental/temporary implementation of device utilizing sshshell connection. It's functionality will be merged into AdbRemote device in Moler 2.0.0 and AdbRemote2 will be removed.
```